samcli/commands/check/command.py

```diff
@@ -0,0 +1,161 @@
+"""
+CLI command for "deploy" command
+"""
+import os
+
+import logging
+import functools
+
+import click
+
+from samcli.cli.main import pass_context, common_options as cli_framework_options, aws_creds_options, print_cmdline_args
+from samcli.cli.cli_config_file import TomlProvider, configuration_option
+from samcli.lib.telemetry.metric import track_command
+from samcli.lib.utils.version_checker import check_newer_version
+from samcli.commands._utils.options import template_option_without_build
+
+from samtranslator.translator.translator import Translator
+from samtranslator.public.exceptions import InvalidDocumentException
+
+import boto3
+from samtranslator.translator.managed_policy_translator import ManagedPolicyLoader
+from samtranslator.parser import parser
+from boto3.session import Session
+from samcli.yamlhelper import yaml_dump
+from samcli.lib.utils.packagetype import ZIP
+
+from samcli.lib.replace_uri.replace_uri import ReplaceLocalCodeUri
+from .exceptions import InvalidSamDocumentException
+
+
+SHORT_HELP = "Checks template for bottle necks."
+
+
+HELP_TEXT = """
+Check your application to determine if any endpoints will not be able to
+provide the expected arival rate of data. You will need to provide the
+expected duration of each lambda function, as well as the expected 
+per-second arrival rate. You will then be informed of the expected cost
+of running this application, as well as any bottle necks that may exist.
+
+This command must be run in the main directory of your application. 
+This command will work on any SAM application. It can also run 
+on a CloudFormation template.
+
+Connections between resources can be made after all required data is
+provided
+"""
+
+CONFIG_SECTION = "parameters"
+LOG = logging.getLogger(__name__)
+
+
+@click.command(
+    "check",
+    short_help=SHORT_HELP,
+    help=HELP_TEXT,
+)
+@configuration_option(provider=TomlProvider(section="parameters"))
+@template_option_without_build
+# options go here with this format
+# @click.option(
+#     "--test",
+#     "-t",
+#     required=False,
+#     is_flag=True,
+#     help="Test number 1",
+# )
+@aws_creds_options
+@cli_framework_options
+@pass_context
+@track_command
+@check_newer_version
+@print_cmdline_args
+def cli(
+    ctx,
+    template_file,
+    config_file,
+    config_env,
+):
+    """
+    `sam deploy` command entry point
+    """
+    # All logic must be implemented in the ``do_cli`` method. This helps with easy unit testing
+    do_cli(ctx, template_file)  # pragma: no cover
+
+
+def do_cli(ctx, template_path):
+    """
+    Implementation of the ``cli`` method
+
+    Translate template into CloudFormation yaml format
+    """
+
+    converted_template = transform_template(ctx, template_path)
+
+    click.echo("... analyzing application template")
+
+
+def load_policies():
```

Comment:
I see that you put your functionality into this `command.py` file/module. I would rather move them into [`samcli/lib`](https://github.com/aws/aws-sam-cli/tree/develop/samcli/lib) with its own module. Then you can keep this file just a bridge between the user inputs and the actual implementation details.